pallets/passkey/src/tests.rs

```diff
@@ -543,3 +547,40 @@ fn pre_dispatch_with_exceeding_weight_should_fail() {
 		assert_err!(v, InvalidTransaction::ExhaustsResources);
 	});
 }
+
+#[test]
+fn passkey_public_key_scale_and_eip712_compatibility_guard() {
+	let public_key = [
+		0u8, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24,
+		25, 26, 27, 28, 29, 30, 31, 32,
+	];
+	let pass_key = PasskeyPublicKey(public_key.clone());
+
+	assert_eq!(pass_key.encode(), pass_key.inner().encode());
+	assert_eq!(pass_key.encode(), pass_key.inner());
+	assert_eq!(pass_key.encode(), public_key);
+}
+
+#[test]
+fn ethereum_eip712_signatures_for_passkey_publickey_should_work() {
+	new_test_ext().execute_with(|| {
+		let payload = PasskeyPublicKey (from_hex("0x40a6836ea489047852d3f0297f8fe8ad6779793af4e9c6274c230c207b9b825026").unwrap().try_into().unwrap());
+		let encoded_payload = payload.encode_eip_712();
+
+		// following signature is generated via Metamask using the same input to check compatibility
+		let signature_raw = from_hex("0xbafaf5e21695a502b2d356b4558da35245aa1be7161f01a5f0224fbfdf85b5c52898fc495ab1ca9b68c3b07e23d31a5fe1686165344b22bc14201f293d54f36b1b").expect("Should convert");
+		let unified_signature = UnifiedSignature::from(ecdsa::Signature::from_raw(
+			signature_raw.try_into().expect("should convert"),
+		));
+
+		// 0x509540919faacf9ab52146c9aa40db68172d83777250b28e4679176e49ccdd9fa213197dc0666e85529d6c9dda579c1295d61c417f01505765481e89a4016f02
```

Comment:
What is the purpose of this comment?

Reply:
That is the non compressed public key of keypair used in metamask for signature. It is not required but it is helpful to have. I'll this in comments